docs/testing.md

```diff
@@ -0,0 +1,146 @@
+---
+title: "Testing"
+date: 2018-05-02T00:00:00+00:00
+weight: 37
+geekdocRepo: https://github.com/owncloud/ocis
+geekdocEditPath: edit/master/docs
+geekdocFilePath: testing.md
+---
+
+
+## Acceptance tests
+
+We are using the ownCloud 10 acceptance testsuite against ocis. To set this up you need the owncloud 10 core repo, a ldap server that the acceptance tests can use to manage users, a redis server for file-versions and the ocis code.
+
+### Getting the tests
+
+All you need to do to get the acceptance tests is check out the core repo:
+```
+git clone https://github.com/owncloud/core.git
+```
+
+### Run a ldap server in a docker container
+
+The ownCloud 10 acceptance tests will need write permission. You can start a suitable ldap server in a docker container with:
+
+```
+docker run --hostname ldap.my-company.com \
+    -e LDAP_TLS_VERIFY_CLIENT=never \
+    -e LDAP_DOMAIN=owncloud.com \
+    -e LDAP_ORGANISATION=ownCloud \
+    -e LDAP_ADMIN_PASSWORD=admin \
+    --name docker-slapd \
+    -p 127.0.0.1:389:389 \
+    -p 636:636 -d osixia/openldap
+```
+### Run a redis server in a docker container
+
+File versions need a redis server. Start one with docker by using:
+
+`docker run -e REDIS_DATABASES=1 -p 6379:6379 -d webhippie/redis:latest`
+
+### Run ocis with that ldap server
+
+`ocis` provides multiple subcommands. To configure them all via env vars you can export these environment variables.
+
+```
+export REVA_USERS_DRIVER=ldap
+export REVA_LDAP_HOSTNAME=localhost
+export REVA_LDAP_PORT=636
+export REVA_LDAP_BASE_DN='dc=owncloud,dc=com'
+export REVA_LDAP_USERFILTER='(&(objectclass=posixAccount)(cn=%s))'
+export REVA_LDAP_GROUPFILTER='(&(objectclass=posixGroup)(cn=%s))'
+export REVA_LDAP_BIND_DN='cn=admin,dc=owncloud,dc=com'
+export REVA_LDAP_BIND_PASSWORD=admin
+export REVA_LDAP_SCHEMA_UID=uid
+export REVA_LDAP_SCHEMA_MAIL=mail
+export REVA_LDAP_SCHEMA_DISPLAYNAME=displayName
+export REVA_LDAP_SCHEMA_CN=cn
+export LDAP_URI=ldap://localhost
+export LDAP_BINDDN='cn=admin,dc=owncloud,dc=com'
+export LDAP_BINDPW=admin
+export LDAP_BASEDN='dc=owncloud,dc=com'
+```
+
+Then you need to start ocis
+```
+bin/ocis server
+```
+
+### Run the acceptance tests
+
+In the ownCloud 10 core repo run
+
+```
+make test-acceptance-api \
+TEST_SERVER_URL=http://localhost:9140 \
+TEST_EXTERNAL_USER_BACKENDS=true \
+TEST_OCIS=true \
+OCIS_REVA_DATA_ROOT=/var/tmp/reva/ \
+BEHAT_FILTER_TAGS='~@skipOnOcis&&~@skipOnLDAP&&@TestAlsoOnExternalUserBackend&&~@local_storage'
+```
+
+Make sure to adjust the settings `TEST_SERVER_URL` and `OCIS_REVA_DATA_ROOT` according to your environment
+
+This will run all tests that can work with LDAP and are not skipped on OCIS
+
+To run a single test add `BEHAT_FEATURE=<feature file>`
+
+### use existing tests for BDD
+
+As a lot of scenarios are written for oC10, we can use those tests for Behaviour driven development in ocis.
```

Comment:
```suggestion
As a lot of scenarios are written for oC10, we can use those tests for Behaviour driven development in OCIS.
```

Reply:
I've made all occurrences lower-case, except of the environment variables of course